File: slicing-android/app/src/main/java/com/crescentflare/slicinggame/infrastructure/appconfig/CustomAppConfigManager.kt
package com.crescentflare.slicinggame.infrastructure.appconfig

import com.crescentflare.dynamicappconfig.manager.AppConfigBaseManager
import com.crescentflare.dynamicappconfig.model.AppConfigBaseModel

/**
 * App config: custom configuration manager
 */
class CustomAppConfigManager : AppConfigBaseManager() {

    // --
    // Static: singleton access
    // --

    companion object {

        var instance = CustomAppConfigManager()

        fun currentConfig(): CustomAppConfigModel {
            return instance.forcedConfigModel ?: instance.currentConfigInstance as CustomAppConfigModel
        }

    }


    // --
    // Member
    // --

    private var forcedConfigModel: CustomAppConfigModel? = null


    // --
    // Implementation
    // --

    override fun getBaseModelInstance(): AppConfigBaseModel {
        return CustomAppConfigModel()
    }

    fun setForcedConfigModel(model: CustomAppConfigModel) {
        forcedConfigModel = model
    }

}

File: slicing-ios/SlicingGame/Infrastructure/AppConfig/CustomAppConfigManager.swift
//
//  CustomAppConfigManager.swift
//  App config integration: manager class to facilitate app config selection and model definition
//

import UIKit
import DynamicAppConfig

class CustomAppConfigManager: AppConfigBaseManager {

    // --
    // MARK: Singleton instance
    // --
    
    static let sharedManager: CustomAppConfigManager = CustomAppConfigManager()

    
    // --
    // MARK: Methods
    // --
    
    override func obtainBaseModelInstance() -> AppConfigBaseModel {
        return CustomAppConfigModel()
    }
    
    static func currentConfig() -> CustomAppConfigModel {
        return sharedManager.currentConfigInstance() as! CustomAppConfigModel
    }

}
